main.py

```diff
@@ -210,7 +210,7 @@ def do_match(cls, flow, args):
             if name not in args:
                 return False
             field = args[name]
-            if name not in ('ipv4_src', 'ipv4_dst', 'ipv6_src', 'ipv6_dst'):
+            if name not in ('ipv6_src', 'ipv6_dst'):
```

Comment:
This conditional here was differentiating any IP related field, aka `ipv6_src`, `ipv6_dst` for IPv6 and `nw_src` and `nw_dst` and then simply performing a match without any mask, however, as it is, it would blow up, fields who are not in `('ipv6_src', 'ipv6_dst')`, such as `dl_src` would probably blow up here since it would be tried to parse as an `ipaddress`. 

Let's take the opportunity and fix this historical convoluted logic here, let's explicitly only try to parse as an `ipaddress.ip_address` if it is an actual ip address, and then you can leave the `if field_flow != field` as a generic checker in the end. See what I mean? That way, over time if newer or more advanved field matches need to be introduced you can introduce future ifs doing something like `match_field_xyz(field, field_flow)` as you've done for `match_field_dl_vlan`, `dl_vlan` neeeded it because it's a bit more complex because it can have a mask (most openflow fields support mask, as flow_manager start to support more with masks then sdntrace_cp will keep improving this part). 

For all fields that `sdntrace_cp` claims to support on its `openapi.yml` we should have unit tests with `do_match(flow, args)` covering that case, you don't have do it in a single PR, but we'll need to cover those fields too. Remember, everything that's on an `openapi.yml` on a stable NApp means that our backend team is supporting it. OK?

Reply:
Ok!